core/src/main/scala/chisel3/Data.scala

```diff
@@ -372,6 +374,12 @@ abstract class Data extends HasId with NamedComponent with SourceInfoDoc {
     }
   }
 
+  // probeInfo only exists if this is a probe type
+  private var _probeInfoVar:      ProbeInfo = null
+  private def _probeInfo:         Option[ProbeInfo] = Option(_probeInfoVar)
+  private[chisel3] def probeInfo: Option[ProbeInfo] = _probeInfo
```

Comment:
Also not sure of the purpose of `_probeInfo` since we also have `probeInfo`
```suggestion
  private[chisel3] def probeInfo: Option[ProbeInfo] = Option(_probeInfoVar)
```

Reply:
I was confused about this, too. I noticed that the same pattern is used for `_bindingVar` with a note about it providing better memory usage.